src/mixins/onThisPageRegistrator.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * This source file is part of the Swift.org open source project
+ *
+ * Copyright (c) 2021 Apple Inc. and the Swift project authors
+ * Licensed under Apache License v2.0 with Runtime Library Exception
+ *
+ * See https://swift.org/LICENSE.txt for license information
+ * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import { SectionKind } from 'docc-render/constants/PrimaryContentSection';
+import { anchorize } from 'docc-render/utils/strings';
+import {
+  MainContentSectionAnchors,
+  PrimaryContentSectionAnchors,
+} from 'docc-render/constants/ContentSectionAnchors';
+
+/**
+ * Crawls the `topicData` of a page, and extracts onThisPage sections.
+ */
+export default {
+  watch: {
+    topicData: {
+      immediate: true,
+      handler: 'extractOnThisPageSections',
+    },
+  },
+  methods: {
+    shouldRegisterContentSection(subSection) {
+      return subSection.type === 'heading' && subSection.level < 4;
+    },
+    extractOnThisPageSections(topicData) {
```

Comment:
Is there a reason why this is done with a watcher and a method as opposed to just a computed property that depends on `topicData`? I assume there is, but I'm just curious.

Reply:
A computed property should not mutate state. In our case we want to extract the sections and save them to the store.

Unless we switch to a scoped slot and provide the data to the child, dropping the store, I think this is the way to go.

Reply:
Yeah, I just meant for calculating the sections themselves, not adding them to the store. Anyways, this is fine as-is—I was just curious.